### PR TITLE
feat: Introduce AnyType

### LINKED ISSUE #6572: Add new internal types `AnyType` and `AnyBool`
... which are used in `Monomorph` when free (unbound) type variables of their kind are encountered.

Functionally `AnyType` is equivalent to `Unit`, but we should keep its type distinct.

I don't think we need `AnyEff` or `AnyRec` etc, because there we will substitute in `Pure`, `EmptyRecord`, and so forth.

### PR DESCRIPTION
**Pull request by JonathanStarup**
fixes #6572

Added after Void and before Unit

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/ast/TypeConstructor.scala [hunk 1]
```diff
@@ -408,4 +408,9 @@ object TypeConstructor {
     */
   case class Error(kind: Kind) extends TypeConstructor
 
+  @IntroducedBy(MonoDefs.getClass)
+  case object AnyType extends TypeConstructor {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/ast/TypeConstructor.scala:412]: You have just  thrown it in here, but I think it belongs next to Void and Unit. I would probably put it between those two.

File: main/src/ca/uwaterloo/flix/language/phase/jvm/JvmOps.scala [hunk 2]
```diff
@@ -70,8 +70,6 @@ object JvmOps {
     case MonoType.Enum(_) => JvmType.Object
     case MonoType.Arrow(_, _) => getFunctionInterfaceType(tpe)
     case MonoType.Native(clazz) => JvmType.Reference(JvmName.ofClass(clazz))
-
-    case _ => throw InternalCompilerException(s"Unexpected type: '$tpe'.", SourceLocation.Unknown)
```
Comment (JonathanStarup) [inline main/src/ca/uwaterloo/flix/language/phase/jvm/JvmOps.scala:74]: This case matches nothing so I deleted it

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/jvm/JvmOps.scala:74]: Neat

File: main/src/ca/uwaterloo/flix/language/phase/MonoDefs.scala [hunk 3]
```diff
@@ -72,7 +72,7 @@ object MonoDefs {
       case Kind.RecordRow => Type.RecordRowEmpty
       case Kind.SchemaRow => Type.SchemaRowEmpty
       case Kind.CaseSet(sym) => Type.Cst(TypeConstructor.CaseSet(SortedSet.empty, sym), tpe0.loc)
-      case _ => Type.Unit
+      case _ => Type.Cst(TypeConstructor.AnyType, SourceLocation.Unknown)
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/MonoDefs.scala:75]: Aaah. But we have the source location!

File: main/src/ca/uwaterloo/flix/language/phase/Resolver.scala [hunk 4]
```diff
@@ -3325,6 +3325,7 @@ object Resolver {
         case TypeConstructor.CaseUnion(_) => Result.Err(ResolutionError.IllegalType(tpe, loc))
         case TypeConstructor.Error(_) => Result.Err(ResolutionError.IllegalType(tpe, loc))
 
+        case TypeConstructor.AnyType => throw InternalCompilerException(s"unexpected type: $tc", tpe.loc)
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:3328]: Move upwards

Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:3328]: You mean remove the empty line above or move anytype up to unit? Both of them is inconsistent with the rest of the function.

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**